### ISSUE TITLE
Inconsistency in handling `self.annotations`

### ISSUE THREAD

**Issue by apdavison** [NeuralEnsemble/python-neo#166]
In the `__array_finalize__` method of `AnalogSignal` and `SpikeTrain`, a default value of `None` is used for the `annotations` attribute. This is inconsistent with the expectation in many other parts of the code that `annotations` be a dict.

I propose to give `annotations` a default value of `{}`.

**Comment by grg2rsr** [comment]
I think this is relevant to this, but I might be mistaken. I just found that performing in place operations on a SpikeTrain removes it's annotations, I assume because the `__array_finalize__` set's it to `None`
e.g.
```python
import neo
import quantities as pq
st = neo.core.SpikeTrain([1,2,3]*pq.s,t_stop=4*pq.s,label='foo')
print(st.annotations) // prints: {'label': 'foo'}
st2 = st - 2 * pq.s
print(st2.annotations) // prints: None
```
for `neo = '0.4.1.1'` and both python2.7 and 3.5

**Comment by apdavison** [comment]
Fixed in #280
